sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -6903,6 +6903,27 @@ public void testMaxSubqueryRows() throws Exception
     );
   }
 
+  @Test
+  public void testZeroMaxNumericInFilter() throws Exception
```

Comment:
W.r.t tests, following case can also be helpful: when the number of IN filters in the query are greater than the MAX_NUMERIC_IN_FILTERS.

Reply:
Yes, we should add a test that covers this case. @somu-imply can you please add one?

Reply:
Added